### ISSUE TITLE
"This Windows version (6.3.9600) does not support the required Bluetooth API. Consider updating to a more recent Windows (100.10586 or above)"

### ISSUE THREAD

**Issue by howtologinquickwiththirtyninecharacters** [vkbo/novelWriter#572]
Running on Windows 8.1, the workaround is to delete `\Lib\site-packages\PyQt5\Qt\bin\Qt5Bluetooth.dll` or rename it.

**Comment by vkbo** [comment]
Yeah, the install package is for Windows 10. Unfortunately, pyinstaller adds all Qt5 libraries. I have a list that strips away the largest ones, but I'll add this one too. It is not needed at all.

**Comment by vkbo** [comment]
Hmm, how did you install this? Via the Windows installer, or did you use `pip install`?

**Comment by vkbo** [comment]
I checked in the Windows installer, and I exclude the bluetooth package. Anyway, the above path seems to indicate it's an issue with the PyQt5 package installed via `pip`. If so, it's not a novelWriter issue, but an issue with https://pypi.org/project/PyQt5/

**Comment by howtologinquickwiththirtyninecharacters** [comment]
> 
> 
> Hmm, how did you install this? Via the Windows installer, or did you use `pip install`?

`git clone` and `pip install` in a virtualenv. I found a few similar issues on other repositories using PyQt5 and [this one](https://github.com/winpython/winpython/issues/592#issuecomment-408290186) seems to point in the right direction but I have no experience with PyQt5 or the code in novelWriter so I don't know what to change.

**Comment by vkbo** [comment]
novelWriter explicitly imports only the tools it needs from PyQt5, and Bluetooth is definitely not one of them. So the suggested solution from your link is already covered. Either there is a dependency loading in the background, or the PyQt5 package has issues on Windows 8.1. I'm haven't been a Windows user since Windows Vista or 7, so I have limited knowledge of the quirks of older Windows versions.

You can always try to install one of the older versions of PyQt5 from https://pypi.org/project/PyQt5/#history

novelWriter isn't very picky on the version, but the higher, the better. But even 5.3 should work.

**Comment by howtologinquickwiththirtyninecharacters** [comment]
Another similar bug report I read said this doesn't happen in Windows 7 or 10, only in 8.1. I don't expect this to be fixed in PyQt5 because it's already a few years old, I think I'll keep using the recommended version and deleting the Bluetooth dll. Thanks for the help and information. Should we close the bug since it is not a novelWriter bug but one of its dependencies' bug?

**Comment by vkbo** [comment]
Sure, I'll close it. :)

**Comment by smaragdus** [comment]
I supposed that [novelWriter](https://novelwriter.io/) would not install on Windows 8 but it did- at least version [1.6.2](https://github.com/vkbo/novelWriter/releases/tag/v1.6.2) installs fine and runs fine under Windows 8.

But after the installation I noticed that [novelWriter](https://github.com/vkbo/novelWriter) was infected with [CODE OF CONDUCT](https://github.com/vkbo/novelWriter/blob/main/CODE_OF_CONDUCT.md) so I uninstalled it immediately.

After the uninstallation I noticed that the uninstaller did not clean the leftovers which remained in AppData:

- C:\Users\User\AppData\Local\novelwriter
- C:\Users\User\AppData\Roaming\novelwriter

So I had to clean up after the uninstaller.

In short- [novelWriter](https://novelwriter.io/) works under Windows 8 but [CODE OF CONDUCT](https://github.com/vkbo/novelWriter/blob/main/CODE_OF_CONDUCT.md) makes it unacceptable to install and use.
